### DISCUSSION TITLE
Can an event be sent to wezterm from a script?

### DISCUSSION THREAD

**Discussion by crispinb** [wezterm/wezterm#3424] [Q&A] [answered]
I have a function in my `wezterm.lua` that responds to an event. Currently I trigger the event from a keybinding (`wezterm.action{EmitEvent=[event name]}`). Can I send that same event from an external script (lua presumably, but the language doesn't matter). That script would be running from the wezterm window that I wish to receive the event.

**Comment by nekowinston** [comment, marked as answer]
There's the event [`user_var_changed`](https://wezfurlong.org/wezterm/config/lua/window-events/user-var-changed.html), where you could call [`wezterm.emit`](https://wezfurlong.org/wezterm/config/lua/wezterm/on.html#custom-events) to propagate it to your own action.

**Reply by wez** [reply]
https://wezfurlong.org/wezterm/recipes/passing-data.html discusses this topic in more detail

**Comment by crispinb** [comment]
Excellent, thank you both. That should do the job.
